### PR TITLE
deal with non-existent node publish paths

### PR DESCRIPTION
**Pull request by WanzenBug**
* Wrong paths return the expected NotFound error in NodeGetVolumeStats
* Remove created mount point file/dir in NodeUnpublishVolume
* Make NodeUnpublishVolume indempotent

### CODE REVIEW

**Comment by WanzenBug** [comment]
Wait for #94 to be merged

File: pkg/client/linstor.go [hunk 1]
```diff
@@ -1124,8 +1125,15 @@ func (s *Linstor) setDevReadWrite(srcPath string) error {
...
+//
+// Non-existent paths return (true, nil).
 func (s *Linstor) IsNotMountPoint(target string) (bool, error) {
-	return s.mounter.IsNotMountPoint(target)
+	notMounted, err := s.mounter.IsNotMountPoint(target)
+	if err != nil && !os.IsNotExist(err) {
+		return false, err
+	}
+
+	return notMounted, nil
```
Comment (rck) [inline pkg/client/linstor.go:1136]: So there is the case where it was an `err`, and where it was an `os.IsNotExist()`, then you return what `IsNotMountPoint` gave you for `notMounted`. Even if that works in this particular implementation, the common rule/"contract" IMO is: if it es an `error`, you can not rely on the other return values what so ever, it is allowed to be anything.

Please rephrase that.

File: pkg/client/linstor.go [hunk 2]
```diff
@@ -1134,18 +1142,23 @@ func (s *Linstor) Unmount(target string) error {
...
+		// Still needs to be unmounted
+		err := s.mounter.Unmount(target)
+		if err != nil {
+			return fmt.Errorf("unable to unmount target '%s': %w", target, err)
+		}
 	}
 
-	err = s.mounter.Unmount(target)
+	// We create a file/directory in Mount() which must be removed again
+	out, err := s.mounter.Run("rm", "-fd", target)
```
Comment (rck) [inline pkg/client/linstor.go:1159]: we are all GNU/Linux specific anyways, here and in LINSTOR and what not, but I'd prefer a "Go solution" if it is possible. Guess that should not be too cumbersome in this case.

**Review by rck — CHANGES_REQUESTED** [review]

File: pkg/client/linstor.go (hunk 2 above)
Comment (WanzenBug) [reply pkg/client/linstor.go]: I wasn't sure if that would always work: there is some magic going on in the `mounter.Run()` method regarding namespaces. But that does not seem to be active in our containers, so a normal `os.Remove()` works :shrug:

File: pkg/client/linstor.go (hunk 1 above)
Comment (WanzenBug) [reply pkg/client/linstor.go]: You are absolutely right on that, updated it just now.

**Comment by rck** [comment]
thanks!

**Pull request merged**